examples/lang-err-integration-tests/call-builder/lib.rs

```diff
@@ -201,35 +244,249 @@ mod call_builder {
+                .await
+                .expect("instantiate failed")
+                .account_id;
+
+            let code_hash = client
+                .upload("constructors_return_value", &ink_e2e::alice(), None)
+                .await
+                .expect("upload `constructors_return_value` failed")
+                .code_hash;
+
+            let selector = ink::selector_bytes!("try_revert_new");
+            let init_value = true;
+            let call =
+                build_message::<CallBuilderTestRef>(contract_acc_id).call(|contract| {
+                    contract.call_instantiate_fallible(code_hash, selector, init_value)
+                });
+            let call_result = client.call(&mut ink_e2e::alice(), call, 0, None).await;
+
+            assert!(
+                call_result.is_err(),
+                "Call execution should've failed, but didn't."
+            );
+
+            let contains_err_msg = match call_result.unwrap_err() {
+                ink_e2e::Error::CallDryRun(dry_run) => {
+                    String::from_utf8_lossy(&dry_run.debug_message)
+                        .contains("Error from the Contracts pallet.: Decode(Error)")
+                }
+                _ => false,
+            };
+            assert!(
+                contains_err_msg,
+                "Call execution failed for an unexpected reason."
+            );
+
+            Ok(())
+        }
+
+        #[ink_e2e::test(additional_contracts = "../constructors-return-value/Cargo.toml")]
+        async fn e2e_create_builder_with_fallible_revert_constructor_encodes_err(
+            mut client: ink_e2e::Client<C, E>,
+        ) -> E2EResult<()> {
+            let constructor = CallBuilderTestRef::new();
+            let contract_acc_id = client
+                .instantiate("call_builder", &ink_e2e::bob(), constructor, 0, None)
+                .await
+                .expect("instantiate failed")
+                .account_id;
+
+            let code_hash = client
+                .upload("constructors_return_value", &ink_e2e::bob(), None)
+                .await
+                .expect("upload `constructors_return_value` failed")
+                .code_hash;
+
+            let selector = ink::selector_bytes!("try_revert_new");
+            let init_value = false;
+            let call =
+                build_message::<CallBuilderTestRef>(contract_acc_id).call(|contract| {
+                    contract.call_instantiate_fallible(code_hash, selector, init_value)
+                });
+            let call_result = client
+                .call(&mut ink_e2e::bob(), call, 0, None)
+                .await
+                .expect(
+                    "Client failed to call `call_builder::call_instantiate_fallible`.",
+                )
+                .value
+                .expect("Dispatching `call_builder::call_instantiate_fallible` failed.");
+
+            assert!(
```

Comment:
I wanna point out that this is technically incorrect, since the dispatch to `try_revert_new` did succeed, and then callee manually put this in the buffer.

The caller would need to make sure that they're not calling contracts which encode invalid values to the error output buffer.